torch/csrc/distributed/c10d/Functional.cpp

```diff
@@ -10,11 +10,25 @@
 
 namespace {
 
+// Forces all functional collectives to wait immediately.
+// Useful for debugging memory leaks with compile that surface
+// due to compile not properly waiting on functional collectives.
+bool force_synchronous_functional_collectives() {
+  static char const* temp = getenv("TORCH_FORCE_SYNCHRONOUS_COLLECTIVES");
```

Comment:
Use std::getenv. Also what if the user wants to explicitly disable it?

Reply:
I'll probably just switch to use the same API's to read envvars as the other distributed code: https://github.com/pytorch/pytorch/blob/main/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L780